otter/integration/tests/test_convergence.py

```diff
@@ -129,6 +178,16 @@ def _choose_half_the_servers(self, (code, response)):
         self.n_killed = self.n_servers / 2
         return ids[:self.n_killed]
 
+    def _choose_random_servers(self, (code, response), n):
+        """Selects ``n`` randomly selected servers from those returned by the
+        ``get_scaling_group_state`` function.
+        """
+
+        if code != 200:
+            raise Exception("Got 404; dude, where's my scaling group?")
```

Comment:
The 404 is accurate; if you look at the code for get_scaling_group_status in the ScalingGroup class, you'll see that it check_results([200, 404]).  200 and 404 are the only "valid" codes it returns.  Anything else is an exception.

So, this issue has come up repeatedly.  I'm not sure why, but it's very annoying.  What can I do to avoid this in the future?